### PR TITLE
fix the input shape to match (:, timesteps, feature) of keras' lstm

### PR DESCRIPTION
**Pull request by Ginurx**
The input shape of an LSTM layer should be a tensor of shape `(:, timesteps, feature)` as specified in the [documentation](https://keras.io/api/layers/recurrent_layers/lstm/).

This change ensures that the input shape matches the expected format so that the LSTM layer processes the inputs recurrently over time steps.

Previously, the code passed inputs with the shape (:, 1, look_back), treating each sample as a `look_back`-dimensional feature vector. This caused the LSTM to function incorrectly.

**Impact of the Fix:**

```
Before the fix:
Train Score: 0.50 RMSE
Test Score: 13.97 RMSE

After the fix:
Train Score: 0.50 RMSE
Test Score: 2.27 RMSE
```

This fix significantly improves the model's performance on the test set, indicating that the LSTM now correctly learns temporal patterns.

### CODE REVIEW

**Comment by omerbsezer** [comment]
Thanks
